kmir/src/kmir/cargo.py

```diff
@@ -95,8 +95,17 @@ def default_target(self) -> str:
 
 
 def cargo_get_smir_json(rs_file: Path, save_smir: bool = False) -> dict[str, Any]:
-    smir_json_result = SMIR_JSON_DIR / rs_file.with_suffix('.smir.json').name
-    run_process_2(['cargo', 'run', '--', '-Zno-codegen', str(rs_file)], cwd=SMIR_JSON_DIR)
+    if Path(IN_TREE_SMIR_JSON_DIR).exists():
+        # prefer local dependency if it exists (i.e., we are in a source/build tree)
+        command = ['cargo', 'run', '--', '-Zno-codegen', str(rs_file)]
+        cwd = IN_TREE_SMIR_JSON_DIR
+    else:
+        # otherwise use 'stable-mir-json' from the path (fail if it does not exist)
+        command = ['stable-mir-json', '-Zno-codegen', str(rs_file)]
+        cwd = Path.cwd()
+
+    smir_json_result = cwd / rs_file.with_suffix('.smir.json').name
+    run_process_2(command, cwd=cwd)
```

Comment:
I think we shouldn't have the "prefer in-tree" solution at all actually. We should let the user decide outside the application how they want to setup their path and stuff, where they can decide which version they prefer.

Reply:
Not a blocker for merging, but I think there's no reason to keep the old logic around.

Reply:
> I think we shouldn't have the "prefer in-tree" solution at all actually. We should let the user decide outside the application how they want to setup their path and stuff, where they can decide which version they prefer.

OK, we can remove that code then. But maybe later, when we have updated documentation, too. I just want the docker installation to work out of the box right now so IMHO the `stable-mir-json` call is the fall-back at the moment. Will follow up removing the in-tree one later.